app/services/ProcessCsvService.scala

```diff
@@ -0,0 +1,273 @@
+              case Some(anError) => Left(anError)
+              case None =>
+                Right(CsvFileContents(sheetName, sequenceOfEithers.map(_.right.get)))
+            }
+          }
+        }
+      )
+    }
+
+  def processFilesNew(callback: UpscanCsvFileData, source: String => Source[HttpResponse, _])(
+    implicit request: Request[_], hc: HeaderCarrier
+  ): List[Future[Either[Throwable, CsvFileSubmissions]]] =
+    callback.callbackData map { successUpload =>
+
+      val sheetName = successUpload.name.replace(".csv", "")
+      val tryValidator: Either[Throwable, (DataValidator, SheetInfo)] = dataGenerator.getValidatorAndSheetInfo(sheetName, callback.schemeInfo)
+
+      tryValidator.fold(
+        throwable => Future.successful(Left(throwable)),
+        setValidator => {
+          val validator = setValidator._1
+          val sheetInfo = setValidator._2
+
+          val futureListOfErrors: Future[Seq[Either[Throwable, Seq[String]]]] = extractBodyOfRequest(source(successUpload.downloadUrl))
+            .via(eitherFromFunction(processRow(_, successUpload.name, callback.schemeInfo, validator, sheetInfo)))
+            .takeWhile(_.isRight, inclusive = true)
+            .runWith(Sink.seq[Either[Throwable, Seq[String]]])
+
+          futureListOfErrors.map { sequenceOfEithers =>
+            sequenceOfEithers.collectFirst { case Left(x) => x } match {
+              case Some(anError) => Left(anError)
+              case None => Right(CsvFileSubmissions(sheetName, sequenceOfEithers.length, successUpload))
+            }
+          }
+        }
+      )
+    }
+
+  //TODO Old version - remove after successful release of large file changes
+  def extractSchemeData(schemeInfo: SchemeInfo, empRef: String, result: Either[Throwable, CsvFileContents])(
+    implicit request: Request[_], hc: HeaderCarrier
+  ): Future[Either[Throwable, (Int, Int)]] = {
+    result.fold(
+      throwable => Future(Left(throwable)),
+      csvFileContents => {
+        Logger.debug("2.1 result contains: " + csvFileContents)
+        Logger.debug("No if SchemeData Objects " + csvFileContents.contents.size)
+        sendSchemeCsv(SchemeData(schemeInfo, csvFileContents.sheetName, None, csvFileContents.contents.to[ListBuffer]), empRef).map { issues =>
+          issues.fold(
+            throwable => Left(throwable),
+            noOfSlices => Right((noOfSlices, csvFileContents.contents.size))
+          )
+        }
+      }
+    )
+  }
+
+  def extractSchemeDataNew(schemeInfo: SchemeInfo, empRef: String, result: Either[Throwable, CsvFileSubmissions])(
+    implicit request: Request[_], hc: HeaderCarrier
+  ): Future[Either[Throwable, (Int, Int)]] = {
+    result.fold(
+      throwable => Future(Left(throwable)),
+      csvFileSubmissions => {
+        sendSchemeCsv(SubmissionsSchemeData(schemeInfo, csvFileSubmissions.sheetName, csvFileSubmissions.upscanCallback, csvFileSubmissions.fileLength), empRef)
+          .map { issues =>
```

Comment:
This is one of those areas of clarity where I had to take a second look to make sure what I was seeing was correct. Naming our value here 'issues' is a little confusing as it doesn't seem to truly reflect what sendSchemeCsv is returning

Reply:
I've slightly changed this about now: there was no reason for the 'noOfSlices' to be created in sendSchemeCsv, so I've moved it into this block. now, sendSchemeCsv returns an Option[Throwable], as such:
```
        sendSchemeCsv(SubmissionsSchemeData(...)
          .map {
            case Some(throwable) => Left(throwable)
            case None =>
              val noOfSlices: Int = ValidationUtils.numberOfSlices(csvFileSubmissions.fileLength, appConfig.maxNumberOfRowsPerSubmission)
              Right(CsvFileLengthInfo(noOfSlices, csvFileSubmissions.fileLength))
          }

```
(the `noOfSlices` is taken out of `sendSchemeCsv`)

also, because of that move, `sendSchemeCsv`'s entire functionality was calling out to `sendSchemeDataCsv`, so I removed it and renamed `sendSchemeDataCsv` to `sendSchemeCsv` (try to say that three times fast)

I'll push it up later so you can see, but just putting it here now in case that raises any questions!
cc @DannyPennington97